lab_1/main.py

```diff
@@ -2,17 +2,27 @@
 Lab 1
 A concordance extraction
 """
-
-
-def tokenize(text: str) -> list:
+import random
+text = open('data.txt', 'r', encoding= 'utf-8').read()
+def tokenize():
     """
     Splits sentences into tokens, converts the tokens into lowercase, removes punctuation
     :param text: the initial text
     :return: a list of lowercased tokens without punctuation
     e.g. text = 'The weather is sunny, the man is happy.'
     --> ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
     """
-    pass
+    raw = text.split()
+    tokens = []
+    for part in raw:
+        cleaned = []
+        for el in part:
+            if el.isalpha():
+                cleaned.append(el.lower())
+        if cleaned != []:
+            tokens.append(''.join(cleaned))
+    return tokens
+tokenize()
```

Comment:
do not call functions here, just write them. call functions in the start.py file